server.js

```diff
@@ -0,0 +1,29 @@
+'use strict';
+
+const express = require('express');
+const bodyParser = require('body-parser');
+const Auth = require('./auth.js');
+
+const app = express();
+app.use(bodyParser.json());
+
+const server = app.listen(process.env.PORT || 8080, () => {
+  const port = server.address().port;
+  console.log(`App now running on port ${port}`);
+});
+
+const auth = new Auth();
+
+app.post('/user/register', async (req, res) => {
+  const credentials = req.body;
+  if (credentials.username && credentials.password) {
+    const result = await auth.registerUser(credentials.username, credentials.password);
+    if (result.success) {
+      res.status(201).json({status: 201, user: credentials.username});
+    } else {
+      res.status(409).json({status: 409, error: result.message});
```

Comment:
Nope. I'd use either `422` or, in rare cases, `418`.

Reply:
Also, the status is not meant to be sent in the view. That's just an HTTP header.

Reply:
418 is the obvious choice

Reply:
haha, I knew you'd like it